### ISSUE TITLE
Horizontal scroll bar does not update when user enters a long line

### ISSUE THREAD

**Issue by john-a-carroll** [robert-strandh/McCLIM#314]
Bug in mcclim-20170725-git

In an interactor pane, if the user enters a line wider than the pane, the horizontal scroll bar does not update to allow the user to scroll to the end of the line. It only updates if the window is resized.

Here is an example that shows this behaviour:

```
(define-application-frame test-frame ()
  ()
  (:pane
    (scrolling ()
      (make-pane 'interactor-pane :scroll-bars nil))))

(run-frame-top-level (make-application-frame 'test-frame))

```
Now enter a line of input wider than the pane. The horizontal scroll bar does not update.

The bug is due to a typo in scroller-pane/update-scroll-bars. Here is a corrected version:

```
(defmethod scroller-pane/update-scroll-bars ((pane scroller-pane))
  (with-slots (viewport hscrollbar vscrollbar) pane
    (let* ((scrollee (first (sheet-children viewport)))
           (scrollee-sr (sheet-region scrollee))
           (viewport-sr (sheet-region viewport)))
      ;;
      (when hscrollbar
        (let* ((min-value (bounding-rectangle-min-x scrollee-sr))
               (max-value (max (- (bounding-rectangle-max-x scrollee-sr)
                                  (bounding-rectangle-width viewport-sr))
                               (bounding-rectangle-min-x scrollee-sr)))
               (thumb-size (bounding-rectangle-width viewport-sr))
               (value (min (- (nth-value 0 (transform-position
                                            (sheet-transformation scrollee) 0 0)))
                           max-value)))
          (setf (scroll-bar-values hscrollbar) ; <- *** typo: was vscrollbar
                (values min-value max-value thumb-size value))))
      ;;
      (when vscrollbar
        (let* ((min-value (bounding-rectangle-min-y scrollee-sr))
               (max-value (max (- (bounding-rectangle-max-y scrollee-sr)
                                  (bounding-rectangle-height viewport-sr))
                               (bounding-rectangle-min-y scrollee-sr)))
               (thumb-size (bounding-rectangle-height viewport-sr))
               (value (min (- (nth-value 1 (transform-position
                                            (sheet-transformation scrollee) 0 0)))
                           max-value)))
          (setf (scroll-bar-values vscrollbar)
                (values min-value max-value thumb-size value)))))))

```

**Comment by dkochmanski** [comment]
looks good, could you make a PR?

**Comment by dkochmanski** [comment]
fixed by #324
